code/VPC-network/security-groups.tf

```diff
@@ -0,0 +1,92 @@
+# security groups to access public (front end) and private (back end)
+# public
+resource "aws_security_group" "Wordpress-front-securitygroup" {
+ 	name = "Wordpress-front-securitygroup"
+ 	description = "front end security group - http and https inbound"
+ 	vpc_id = aws_vpc.wordpress-vpc.id
+ 	tags =	{
+ 		Name = "Wordpress-front-securitygroup"
+ 		}
+	ingress {
+		from_port = 80
+		to_port = 80
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+		ingress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+	ingress {
+		from_port = "22"
+		to_port = "22"
+		protocol = "tcp"
+		cidr_blocks = [var.MYIP]
+		}
+	egress {
+		from_port = 0
+		to_port = 0
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
```

Comment:
We need to have this rule to allow all outbound traffic? Could we limit that only for database port (3306) and maybe HTTP/S for system updates?

Reply:
we can for sure.
i am still getting my head around NACL vs security groups.
so front end security group would be applied to the app servers. the idea was they could receive inbound https from the internet, ssh from home IP, but outbound anywhere. we can limit to just HTTPS outbound, though i (probably incorrectly!) assumed outbound was less of a concern than inbound.
i could be well off on this though! but that was my thought process.

Reply:
I'd recommend to not stress too much around NACLs and SG rules for now.. 
Even being a requirement for the final solution that all SGs and NACLs only have the required allow rules, for the moment,it's better to have more loose rules as you guys can test things out. Be sure, you won't get all rules ready in this first round.

Reply:
It is a good point, we could relax it know and improve the security later.

One suggestion is to create a backlog item to track it. So we will not take risk to miss that at the end.